lecture-notes/2-vcs.md

```diff
@@ -1940,37 +1942,102 @@ After you have rebased, you can safely push your (renewed) local commits to the
 remote. That is, unless someone else has beaten you to the punch and pushed
 even more commits to the remote. In that case, keep fetching and rebasing.
 
-Note that this re-applying of local commits on top of remote changes
-constitutes *rewriting history*. Some people find this very distasteful and
-even ban it in their projects, preferring merge commits exclusively. The course
-staff has a more moderate view of things: rewriting history is fine if it's a
-private, short-lived development branch. Otherwise, merge.
+> Note that this re-applying of local commits on top of remote changes
+> constitutes *rewriting history*. As we have discussed before, some people
+> find this very distasteful and even ban it in their projects, preferring
+> merge commits exclusively. The course staff has a more moderate view of
+> things: rewriting history is fine if it's a private, short-lived development
+> branch. Otherwise, merge.
 
 #### Merge
 
 It's also possible to use `git merge` to reconcile local and remote changes.
-Merge also comes with ability to fast-forward but that must be requested by
-default with `git merge --ff`.
+Like rebase, merge also comes with ability to fast-forward, but that does not
+happen automatically; it must be requested with `git merge --ff` or
```

Comment:
It definitely does happen automatically: `--ff` is the default. `--ff-only` is not the same as `--ff` and means that the merge should fail if a fast-forward isn't possible.